### ISSUE TITLE
Inconsistent treatments for "similar" percepts 

### ISSUE THREAD

**Issue by Rezenders** [jason-lang/jason_ros#38]
Perception msg 1 is added in the belief base as a Term and perception msg 2 is added as a String. Why is this happening? Is this the expected treatment? I think they should be both added as Term.

Percept msg 1:

```
header: 
  seq: 679
  stamp: 
    secs: 0
    nsecs:         0
  frame_id: ''
agent_name: ''
perception_name: "diagnostics"
parameters: 
  - '[[[energy, 6.208817140973544e-10]]]'
update: True
```

Perception manifest 1:

```
[diagnostics]
name = diagnostics
msg_type = diagnostic_msgs/DiagnosticArray
dependencies = diagnostic_msgs.msg
args = status[](values[](key, value))
buf = update
```

Percept msg 2:

```
header: 
  seq: 33
  stamp: 
    secs: 0
    nsecs:         0
  frame_id: ''
agent_name: "metacontrol"
perception_name: "diagnostics"
parameters: 
  - '[[[Received packets:, 8867], [Dropped packets:, 0], [Buffer overruns:, 0], [Parse
    errors:, 0], [Rx sequence number:, 215], [Tx sequence number:, 0], [Rx total bytes:,
    286009], [Tx total bytes:, 17200], [Rx speed:, 7497.000000], [Tx speed:, 532.000000]],
    [[Satellites visible, 10], [Fix type, 6], [EPH (m), 1.21], [EPV (m), 2.00]], [[Heartbeats
    since startup, 45], [Frequency (Hz), 1.000003], [Vehicle type, Submarine], [Autopilot
    type, ArduPilot], [Mode, MANUAL], [System status, Active]], [[Sensor present, 0x1330FD2F],
    [Sensor enabled, 0x13209D2F], [Sensor health, 0x13309D2F], [3D gyro, Ok], [3D accelerometer,
    Ok], [3D magnetometer, Ok], [absolute pressure, Ok], [GPS, Ok], [laser based position,
    Ok], [3D angular rate control, Ok], [attitude stabilization, Ok], [yaw position,
    Ok], [motor outputs / control, Ok], [AHRS subsystem health, Ok], [Logging, Ok],
    [Battery, Ok], [pre-arm check status. Always healthy when armed, Ok], [CPU Load
    (%), 0.0], [Drop rate (%), 0.0], [Errors comm, 0], [Errors count #1, 0], [Errors
    count #2, 0], [Errors count #3, 0], [Errors count #4, 0]], [[Voltage, 12.59], [Current,
    0.0], [Remaining, 90.0]], [[Timesyncs since startup, 455], [Frequency (Hz), 10.000050],
    [Last RTT (ms), 5.155978], [Mean RTT (ms), 5.670888], [Last remote time (s), 958.590668000],
    [Estimated time offset (s), 1639747256.374345064]]]'
update: True
```

Perception manifest 2:

```
[diagnostics]
name = diagnostics
msg_type = diagnostic_msgs/DiagnosticArray
dependencies = diagnostic_msgs.msg
args = status[](values[](key, value))
buf = update
```

**Comment by Rezenders** [comment]
The issue is occurring in the RosArch perceive method when there are special characters such as space, ":", etc in the parameter string. The parseTerm function raises an exception and then the parameter is interpreted as a String. A solution is to verify if the parameter is a vector and then try to apply parseTerm one by one to each element of the vector instead of applying to the vector as a whole.

```
for (String param : perception.getParameters()) {
  try {
    Term p = parseTerm(param);
     if (p.isVar()) {
      p = new StringTermImpl(param);
    }
    bel.addTerm(p);
  } catch (ParseException e) {
    bel.addTerm(new StringTermImpl(param));
  }
}
```

**Comment by Rezenders** [comment]
But what if the parameter is indeed a string and the string contains a structure similar to a vector? 

I guess in the current version it is interpreted as term if possible and as string if not. So it would not change. May be interesting to force a string interpretation. I just think the java codes should be decoupled from the manifests, maybe send something in the perception msg?

**Comment by Rezenders** [comment]
Fix at 703136d196e89f3d67dacc260b52d3b2ced009ed
